c10/cuda/test/build.bzl

```diff
@@ -17,9 +17,11 @@ def define_targets(rules):
         target_compatible_with = rules.requires_cuda_enabled(),
         deps = [
             "//c10/cuda",
-            "@com_google_googletest//:gmock",
```

Comment:
Instead of doing that, just add `gtest_deps` parameter to `define_targets` and pass one deps from TARGETS file and another from BUILD.bazel